### ISSUE TITLE
Enable optional HTTP proxy

### ISSUE THREAD

**Issue by icereed** [robinpellegrims/pellegrims#176]
**Is your feature request related to a problem? Please describe.**
We need to run NX behind a corporate proxy. Hence, we get the following error because the S3 SDK is not aware
of any proxy:
```
------------------------------------------------------------------------------
Warning: Failed to check if cache file exists in S3
Hash: ae1f1571dbd4dafa3b5d878ddb2e714ddb22824ab364037a7489f3dc0a9a53e5
Error: getaddrinfo ENOTFOUND nx-poc.s3.eu-central-1.amazonaws.com
------------------------------------------------------------------------------
Error: getaddrinfo ENOTFOUND nx-poc.s3.eu-central-1.amazonaws.com
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:71:26) {
  errno: -3008,
  code: 'ENOTFOUND',
  syscall: 'getaddrinfo',
  hostname: 'nx-poc.s3.eu-central-1.amazonaws.com',
  '$metadata': { attempts: 1, totalRetryDelay: 0 }
}
```

**Describe the solution you'd like**
The AWS SDK shall consider HTTPS_PROXY and NO_PROXY env variables.

**Describe alternatives you've considered**
This library of AWS should help: https://github.com/awslabs/aws-sdk-v3-js-proxy.
However, this library does not respect NO_PROXY. Thus, we would need a config flag/ENV to conditionally
enable the proxy.

**Comment by robinpellegrims** [comment]
HTTP_PROXY/HTTPS_PROXY support looks like a useful new feature we could definitely add using [aws-sdk-v3-js-proxy](https://github.com/awslabs/aws-sdk-v3-js-proxy). 

If I understand correctly, the NO_PROXY env variable is used to bypass the proxy set in HTTP_PROXY or HTTPS_PROXY for certain ip's/hostnames. As our nx-remotecache-s3 task runner is only connecting to the S3 bucket endpoint, am I missing a use case that would need the NO_PROXY support?

Would you like to provide a PR?

**Comment by icereed** [comment]
NO_PROXY could be interesting if you want to exclude S3 from proxy but use the proxy for STS or similiar.
If you want I can try a PR, but I would need to wait a bit. I hope I got some time in December.

**Comment by robinpellegrims** [comment]
Oh ok, I see! Let's support both env variables then. 

Feel free to provide a PR whenever you have time. In the meantime i'll see if I can add some unit tests to increase code coverage and make sure we're not breaking anything.

**Comment by robinpellegrims** [comment]
Released in [7.1.0](https://github.com/robinpellegrims/pellegrims/releases/tag/nx-remotecache-s3-7.1.0) thanks to @mscharp
